### DISCUSSION TITLE
/articles/sharepoint-framework-how-to-use-react-hooks-with-spfx-projects/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#240] [Comments]
Learn how to convert a default class-based React component to a functional component, and then see how to implement React Hooks in the web part!
https://www.andrewconnell.com/articles/sharepoint-framework-how-to-use-react-hooks-with-spfx-projects/

**Comment by ZzyzxIsAlreadyTaken** [comment]
you have currentSiteUrl in the code here, but it is not fetched anywhere? Also TypeScript complains about promises not beeing in a try catch.

**Reply by andrewconnell** [reply]
Thanks for bringing this to my attention... I'll refresh the code in the post. The original version of the post predated when Microsoft [revamped their configuration of ESLint in SPFx projects in SPFx v1.15](https://www.voitanos.io/blog/sharepoint-framework-v1-15-and-the-attack-of-eslint/). The changes to the ESLint rules are what's causing the lining warning about promises not containing a `try-catch`.

*BTW, that's not TypeScript complaining about it... that's ESLint*

I'll get the code & post updated for the current version in the coming days.

**Reply by andrewconnell** [reply]
@ZzyzxIsAlreadyTaken Just updated the article... the updated code should be reflected in the next few hours once the CDN caching gods release their grips... thanks again!
